src/transport.toit

```diff
@@ -3,19 +3,102 @@
 // found in the LICENSE file.
 
 import .packets
+import reader
 
 /**
 The backing transport for the MQTT client.
 */
-interface Transport:
+interface Transport implements reader.Reader:
   /**
-  Send a packet to the peer.
+  Write to the transport.
+
+  Returns the number of bytes written.
+  */
+  write bytes/ByteArray -> int
+
+  /**
+  Receives bytes from the peer.
+  */
+  read -> ByteArray?
+
+  /**
+  Closes the transport.
+
+  If another task is sending or receiving, that operation must throw.
+  Any future $write or $read calls must throw.
+
+  The close operation itself must not throw.
+  // TODO(florian): should we deal with disconnections that can throw?
```

Comment:
It feels a little bit odd to have a // inside the /** */.

Reply:
removed.
done.